lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registration_date,
+              :collected_fees,
+              :registered_vehicles
 
-  def initialize(name, address, phone)
+  def initialize(name:, address:, phone:)
     @name = name
     @address = address
     @phone = phone
     @services = []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+    @registered_vehicles << vehicle
+    vehicle.registered_date
+    vehicle.assign_plate
+    add_collected_fees(vehicle)
```

Comment:
nice, I like this helper method you created!

Reply:
Does this negate what you said earlier about including the assign date to the  register method?